run.sh

```diff
@@ -1,36 +1,74 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-waitUrl="`pwd`/ui/waiting-room.html";
+set -e
 
-# Ensure dependencies are installed.
-echo "* Checking dependencies..."
-hash tsc 2>/dev/null
-if [ $? -ne 0 ]; then
-  echo "Please install the typescript compiler with ('sudo npm install -g typescript') before continuing."
-  exit
-fi
-hash multirust 2>/dev/null
-if [ $? -ne 0 ]; then
-  echo "Please install multirust with ('./install-multirust.sh') before continuing."
-  exit
-fi
+waitUrl="$(pwd)/ui/waiting-room.html";
+rustVersion="nightly-2015-08-10"
+debugFlag=false
+openBrowserFlag=false
 
-pushd .
-  # Try using the typescript compiler (tsc) to compile UI
-  echo "* Compiling Editor..."
-  cd ui
+# Parse command line options.
+while test $# -gt 0; do
+case "$1" in
+  -h|--help)
+    echo "Compile and run the Eve editor"
+    echo ""
+    echo "Usage:"
+    echo "    run.sh [options]"
+    echo ""
+    echo "Options:"
+    echo "    -h, --help            Print this message"
+    echo "    -d, --debug           Debug build"
+    echo "    -o, --open-browser    Open editor in default browser"
+    exit 0
+    ;;
+  -d|--debug)
+    debugFlag=true
+    shift
+    ;;
+  -o|--open-browser)
+    openBrowserFlag=true
+    shift
+    ;;
+  *)
+    break
+    ;;
+esac
+done
+
+# Ensure that dependencies are installed.
+printf "* Checking dependencies..."
+deps="tsc multirust"
+for dep in $deps; do
+  if ! which "$dep" &> /dev/null; then
+    printf "  x Please install %s: " "$dep"
```

Comment:
@joshuafcole what do you think about this? 

<img width="547" alt="screen shot 2015-08-25 at 5 13 12 pm" src="https://cloud.githubusercontent.com/assets/10385848/9482780/9d530f90-4b4c-11e5-855d-1544fb29e974.png">